flows/actions/send_msg.go

```diff
@@ -191,52 +202,42 @@ func getTemplateMsg(action *SendMsgAction, run flows.Run, urn urns.URN, channelR
 		}
 
 		evaluatedParams["body"] = evaluatedVariables
-	}
 
-	if action.Templating.Params != nil {
+	} else if a.Templating.Params != nil {
+		for comp, compParams := range a.Templating.Params.Values {
+			localizedCompParams, _ := run.GetTextArray(uuids.UUID(a.Templating.Params.UUID), comp, compParams, nil)
+			evaluatedCompParams := make([]string, len(localizedCompParams))
 
-		for compKey, compVars := range action.Templating.Params.Values {
-			var evaluatedComponentVariables []string
-			if strings.HasPrefix(compKey, "button.") {
-				qrIndex, err := strconv.Atoi(strings.TrimPrefix(compKey, "button."))
+			for i, variable := range localizedCompParams {
+				sub, err := run.EvaluateTemplate(variable)
 				if err != nil {
 					logEvent(events.NewError(err))
 				}
-				paramValue := evaluatedQuickReplies[qrIndex]
-				evaluatedComponentVariables = []string{paramValue}
```

Comment:
I don't get this at all...

Reply:
I think users expected quick replies on the node to be used on the WA templates, so if the template has button that need variables we use the quick replies values set on the node, similarly for attachments

Reply:
But we've never supported this because we've never supported templates with buttons..

Reply:
I think that is what we want to add support for.
The confusion user have been having is that the expect the quick replies to be used for templates.
We have had to clarify that we do not yet support those components yet.

We could also force the user so always specify the parameter for the templates in addition to the quick replies

Reply:
I think when we support button components it will be more obvious to the user. But it's important that the preview message we generate in this case is similar to what will be sent via the WhatsApp template. There's no point adding quick replies to the preview message if the template doesn't use buttons.

Currently we only have `Content()` which is the body text... I think we need the button texts so we can perform `{{1}}` variable substitution on those... but let's worry about that later.